src/main/kotlin/mate/academy/ArrayList.kt

```diff
@@ -1,44 +1,155 @@
 package mate.academy
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays
+import java.util.NoSuchElementException
+
+
 class ArrayList<T> : List<T> {
+
+
+
+    private val defCapacity = 10
+    private val empty_arrayData = arrayOfNulls<Any>(0)
+
+
+    private var array: Array<T?> = arrayOfNulls<Any>(defCapacity) as Array<T?>
+    private var size = 0
+    private var cursor = 0
+
+
+    private fun errorMsg(index: Int): String {
+        return "Index: $index, Size: $size"
+    }
+
+
+    private fun grow(): Array<T?> {
+        val oldCapacity = array.size
+        val multiplicator = if (oldCapacity == size) 1.5 else 1.0
+        val newCapacity = (oldCapacity * multiplicator + 1).toInt()
+        array = if (oldCapacity > 0 || array !== empty_arrayData) {
+            return Arrays.copyOf(array, newCapacity)
+        } else {
+            return arrayOfNulls<Any>(defCapacity) as Array<T?>
+        }
+        return array
+    }
+
+    companion object {
+        fun listToArray(list: List<*>): Array<Any?>? {
+            if (!list.isEmpty()) {
+                return Array(list.size()) { index -> list[index] }
+            }
+            return null
+        }
+    }
+
     override fun add(value: T?) {
-        // implementation
+        val index = cursor
+        this.add(value, index)
+        cursor = index + 1
     }
```

Comment:
No need to save curson into temp index
Could be 
this.add(value, cursor)
cursor = cursor + 1